### ISSUE TITLE
A/V coordinating (Venue A/V details, etc...)

### ISSUE THREAD

**Issue by dcwalk** [ournetworks/2019#33]
Edit: Turned this top-level issue into a tracking one for both TODOs and current thinking on equip.

## Todos 

- [x] Initial venue walkthrough and confirmed site details
- [x] Hank to propose AV recording setup (**July 1st**)
  - [x] Initial draft 
  - [x] Feedback/discussion
  - [x] Revised draft
- [x]  IPFS livestream code changes
  - [ ]  Committed: Pure nginx - tomeshnet/ipfs-live-streaming#76
  - [ ]  Committed: Improve player stream selection UX (e.g. default to one stream if other is not available) --
  - ~~Stretch: Chunk playlist archives at each hour~~ - Dont need to anymore
  - [ ]  Stretch: Scrollback from live player - tomeshnet/ipfs-live-streaming#87
  - [ ]  Stretch: Tagging chunks live and tweet (stretch goal) -- tweet button on player ournetworks.ca only change - 
    - [ ] Ability to start stream at specific time - tomeshnet/ipfs-live-streaming#88
- [x] **@ASoTNetworks** get capture cards to verify
- ~Hank to produce a video ahead of time to get time estimate on post-processing~
- [x] **organizers** Recruit AV helpers (way before event), handle with volunteers
- [x] A/V team Come up with roles (before event)
    - [ ] Create teaching doc
- [x] Mid-July design jam on AV graphics and audio (**August**)
- [x] Hank to produce stream and video graphics (**August**)
- [x] Consensus between organizers and video team on expectations ahead of conference (quick chat after walkthru in **August**)
- [x] Discuss Our Networks coverage of Alexandria (broken out task from this: #109)
- [x] Discuss Filming policy for 2019) -> #105 

----

## A/V Hardware Tracking

Access:  [Own|Borrow|Rent|Buy]

| Equipment | Description | Cost | Access* | Provider |
|---------------|----------------|-------|------------|------------|
| ✅ Audio Mixer         | ?? | $0 | Borrow | TMAC |
| ✅ Microphones | **x4**  ??  | $0 | Borrow | TMAC |
| ✅ Lectern      | ??  | $0 | Borrow | TMAC |
| ✅ Speakers   | ??  | $0 | Borrow | TMAC |
| ✅ Screens    | **x3** 70” screens | $X | Rent 💰 | TMAC  | 
| ✅ Tripod | Steel camera tripod or video tripod  | $0     | Own or Rent | @Shrinks99 <br /> (backup: @dcwalk or Sheridan College) |
| Camera     | Canon XA10 Camcorder <br /> (backup:  )   | $140 <br /> (backup: $0 )  | Rent  💰       | Sheridan / York <br/> (backup: Vistek)  |
| ✅ Capture Cards          | Elgato HD60 / Avermedia GL310 LGP Lite | $0     | Own         | @ASoTNetworks & @dcwalk   |
| HDMI Splitter          | Keliiyo 1x4 active HDMI splitter       | $30    | Buy 💰       | CanadaComputers                                           |
| HDMI Monitor           | x2 Blackmagic Video Assist Monitor     | $210   | Rent   💰      | Vistek                                           |
| SD Cards               | x4 SanDisk Ultra 128GB microSDXC UHS-I | $108   | Buy  💰        | CanadaComputers                                           |
| ✅ Streaming Computer           | ??                                     | $0     | Own         | @ASoTNetworks's laptop                           |
| ~Recording Computer~ | | $0 | |
| HDMI cables            | x2 25ft HDMI cables                    | $37.06   | Buy  💰       | CanadaComputers                                           |
| Offline Audio Recorder | Zoom H4N                               | $0     | Rent        | Sheridan College / York University               |

### Access

- [TMAC: Toronto Media Arts Centre](https://www.tomediaarts.org/space-rentals)
- [CSV: Charles Street Video](https://charlesstreetvideo.com/production_shooting.php)
- [LIFT: Liasion of Independent Filmmakers](https://lift.ca/equipment) Note: we'd have to credit them in videos/website
- [Vistek](https://www.vistek.ca/rentals/)

**Comment by benhylau** [comment]
cc. @ASoTNetworks @Shrinks99 @DarkDrgn2k

**Comment by Shrinks99** [comment]
## Our Networks Stream / AV Wishlist for 2019

### Required Feeds
- Presenter's laptop screen video feed
- Presenter camera video feed
- Presenter microphone audio feed
- Presenter's laptop audio feed

In addition to the live stream having all of these feeds recorded separately would be super ideal.  This would make the post process much easier and allow me to create the same production quality as last year with a fraction of the time.

### Required Hardware to Make That Happen
- Mixer (I actually own two of these now, one is only a microphone mixer. If we don't have a better setup available I'd be happy to lend mine for the two days!)
- Two capture cards (Presenter video & presenter laptop video, same as last year)
- Camera
- Stream computer

### Things That Would Be Neat to Have
- TWO Cameras :0  Having the second angle ad Mozilla was nice sometimes.
- Room tone microphone

Please remind me if I've missed anything and it will be added to the list.

**Comment by ASoTNetworks** [comment]
I have contacted the media team that might be able to help out but they are only available on September 20 and 22 but not on 21.

Their pricing for the 2 days is $700 with 2 cameras.

**Comment by benhylau** [comment]
Adding notes from our [call on June 14](https://hackmd.io/eoE6-cpQSbim67i_NquNHQ?edit). @dcwalk  think these should go in https://github.com/ournetworks/2019/tree/master/notes or another place? I imagine we'd have future AV calls.

**Comment by Shrinks99** [comment]
Alrighty!  Here's the beginnings of a budget.  Could they be more detailed?  Some parts maybe but hopefully this is enough to go off of for right now.  Let me know if you have any questions!

[https://cryptpad.fr/code/#/2/code/view/5DCjYZYq9CtcKezlgskq8BE7QBj9oZMT447rPW-06e0/present/](https://cryptpad.fr/code/#/2/code/view/5DCjYZYq9CtcKezlgskq8BE7QBj9oZMT447rPW-06e0/present/l)

**Comment by benhylau** [comment]
I think this proposal lands _very_ far from what was discussed on our last call and from what I'd expect Our Networks to adopt. For example, starting with 4 capture cards to separate out the recording and streaming chains seem unnecessary, it makes more moving pieces and puts $1000 to $3000 on the budget. We already have access to an Elgato and an AVerMedia card, that should be the starting point? Also, what problem we had last year does this chain splitting try to solve? We had AV out of sync problems, and some video colour / contrast issues that were due to venue equipment, I am not sure what purpose does this $3000 serve.

On the call, we discussed independent recording of the audio and video streams. I see that in the diagram, but not cost information relating to that. We also discussed to propose a low budget plan, but there seems to be low budget options per item, but not an overall plan.

At this stage of the planning, I think knowing the budget is what's most time sensitive. This proposal does not give indication of a reasonable AV budget or what a a low budget alternative looks like.

**Comment by dcwalk** [comment]
@Shrinks99  I couldn't edit the pad so I've made a table to make clearer the specific asks and need at the top of this gh issue: https://github.com/ournetworks/2019/issues/33#issue-423542897

--

I'll also just paste my comments from the Mesh TV chat here for posterity:

Hey Hank , thanks for posting. I'm trying to make sure I understand the math here... the cheapest way you see running this is $1000 for gear, with the most expensive is $3650. And the model you are recommending is purchasing of all of this gear (for insight into our total budget -- that ranges to a recommendation from 10% to over 30% of our current budget).

I was also hoping to see more specific places/gear, for instance, could you recommend the camera you are thinking of buying?

And also, to make sure I understand, why do we need 4 capture cards? (b/c 2 video: deck+video x2). This seems... cumbersome? Are there no solutions that manage this that are in range with the prices you are quoting? ($660 - 850)

also it would be helpful if you linked to where you see us purchasing this gear from!
I think we may have some of this gear ready to hand if we knew specs, for instance the computers you are thinking of (and maybe camcorders too). Can you add them?
Final thought: have you looked at Vistek for pricing rentals? https://www.vistek.ca/rentals/**

**Comment by Shrinks99** [comment]
I am continuing to provide budget options per item as it does not make sense to separate the entire thing into pricing tiers.  Making sweeping decisions will likely not result in effective spending in this situation whereas per-component pricing will allow us to allocate money where it will make the most difference.  Separating outputs to offline VS livestream will resolve the rather creative editing that I had to do in order to fix livestream mistakes in post.  This ensures that can cut to the content that is the most relevant in post even if we don't make the best decisions for the live broadcast, last year I was only able to cut away to slides, this year I will be able to cut between slides and the presenter or both whenever the shot calls for it.  TL;DR Better & faster post process!  With that in mind @ASoTNetworks and I have been able to cut down the hardware pricing to something more reasonable and this is why I like feedback!

# Revised Our Networks AV Equipment List

A revision of [this hardware list](https://cryptpad.fr/code/#/2/code/view/5DCjYZYq9CtcKezlgskq8BE7QBj9oZMT447rPW-06e0/present/) - While AV gear still isn't cheap we have tried to spec out the most cost effective options for this setup.

## HDMI Splitter

Reducing the amount of computer's required to one and using the Blackmagic Video Assist Monitor for recording means that we only need to purchase one HDMI splitter for the screen instead of both for the screen and camera.  This will allow the presenter video to output to the three presentation screens as well as to all AV related gear.

KELIIYO has a 1x4 active HDMI splitter for $20 which is the [listed price on Amazon](https://www.amazon.com/Splitter-Keliiyo-Powered-Converter-Resolutions/dp/B076JDXPRG), this does not include the $10 for shipping.

**TOTAL: $30 (Shipping included)**

## Capture Cards

Because last years capture cards (the Avermedia and Elgato cards) can be made available there is no additional cost to include both of these.

ELGATO HD60: $0

AVERMEDIA GL310 LGP LITE: $0

**TOTAL: $0**

### Offline Audio Recorder

As stated in the other hardware list, I (@Shrinks99) have access to a Zoom H1N recorder which also includes an additional shotgun mic. & mic. stand.  If not available this item can be rented.

**[3 DAY RENTAL: $43.75](https://www.vistek.ca/rentals/details/R303677/zoom-h4next-handy-recorder)**

**Zoom H4N SHERIDAN RENTAL: $0**

## Offline Video Recorder

In leu of a seperate computer setup the Blackmagic Video Assist Monitor seems to be the best way of capturing both the camera and presentation feeds.  This will have to be rented as it is much more cost effective to do so.

**[2x 3 DAY RENTAL: $210](https://www.vistek.ca/rentals/details/R305499/blackmagic-design-video-assist-5-1080p-monitor)**

## Tripod

Sheridan's larger video tripod is required if we go with the Canon XA10 or the SONY PMW-EX1.

**VIDEO TRIPOD SHERIDAN RENTAL: $0**

**@Shrinks99 TRIPOD: $0**

**@dcwalk TRIPOD: $0**

## Camera

Turns out this one is much easier to validate than I previously thought, Atomos offers a nice breakdown of [cameras that offer clean HDMI output](https://www.atomos.com/cameras)!  While these options definately come at different price ranges _all_ should be suitable for our needs.  Any DSLR or camcorder that is capable of being left on and outputs clean 1080p live video over HDMI and has a decent lens available (no super wide angles pls) is good enough for this purpose.

**[SONY PMW-EX1 1 WEEK LIFT RENTAL: $200](https://lift.ca/gear/gear.php?id=183) (Must be a LIFT member)**

**[CANON XA10 3 DAY VISTEK RENTAL: $140](https://www.vistek.ca/rentals/details/R304081/canon-xa10-avchd-camcorder)**

**[CANNON HF S10 YORK RENTAL](http://digital-locker.design.yorku.ca:8080/tech_info/borrow-equipment) (Still needs to be tested for clean output): $10**

**CANNON 7D SHERIDAN RENTAL (Still needs to be tested for clean output): $0**

## Streaming PC

As mentioned previously having to tear down the system every day and set up wasn't ideal however if anything should be cut to divert costs to other areas it should be this one as @HeavyMetal has offered to make his laptop available again this year.  Any official discussion with FreeGeek has yet to happen, that option should be considered to be in the pre-preliminary stages of being sorted out.  Encoding is CPU based, any computer with a 5th gen or more recent i5 or i7 should do a great job although non-rental options should be tested to ensure they work beforehand..

For rental equipment at least one extra day must be added to account for software install and setup costs.

**@ASoTNetworks LAPTOP: $0**

**POSSIBLE FREEGEEK OPTION: $0-$200** (Depending on available hardware and if something needs to be upgraded in a worst case scenario where nothing they have meets the requirements of the encoder, I see this as a very unlikely scenario)

**[ZBOOK STUDIO G3 LAPTOP 3 DAY RENTAL: $312.50, 1 WEEK RENTAL: $375](https://www.vistek.ca/rentals/details/R305876/hewlettpackard-15-zbook-studio-g3-laptop)**

## Cables & Misc

The video mixer goes in the "nice to have" category as it would allow us to display a conference logo on the screens when nobody is presenting and also resolve any resolution or hot-plugging issues that may come up with different screen resolutions being used throughout the conference.

**[2x 25ft HDMI cables: $37.06](https://www.monoprice.com/product?p_id=3992)**

**[Video Mixer - 3 day Vistek rental: $315](https://www.vistek.ca/rentals/details/R305779/roland-vr4hd-hd-av-mixer)**

**Additional extention cords (@Shrinks99): $0**

## TOTALS

Absolute minimum: ~$277

Absolute maximum: ~$1105

**Comment by ana0** [comment]
I have two pretty long HDMI cables and an HDMI splitter than I accidentally stole from a conference recently. I may need to return them before then, but it's also possible we can use

**Comment by Shrinks99** [comment]
@ana0 The HDMI cables just need to be long enough to reach the AV table, I figure 25 feet to be safe (having never seen the venue space) but as long as they reach that’s fine.  I also forgot to include the 4 SD cards we need for this setup because I’m a bonehead so I’ll edit this tonight or @ASoTNetworks will find out how much they are today.

**Comment by ASoTNetworks** [comment]
@Shrinks99 suggest we use 4x 128G SD cards and swap them after each talk.

https://www.canadacomputers.com/product_info.php?cPath=11_1216_175_176&item_id=123610
$29.99  x4 ($119.96)

https://www.amazon.ca/SanDisk-Ultra-microSDXC-Adapter-SDSQUAR-128G-GN6MA/dp/B073JYC4XM/ref=sr_1_3?keywords=128GB+UHS-I&qid=1562340923&s=gateway&sr=8-3
$27.00 x4 ($108)

**Comment by Shrinks99** [comment]
Here's the current table as mentioned in the chat @dcwalk.

| *Equipment*            | *Description*                          | *Cost* | *Access*    | *Provider*                                       |
|------------------------|----------------------------------------|--------|-------------|--------------------------------------------------|
| Mixer                  | ??                                     | $0     | Borrow      | TMAC                                             |
| Microphones            | x4 ??                                  | $0     | Borrow      | TMAC                                             |
| Lectern                | ??                                     | $0     | Borrow      | TMAC                                             |
| Speakers               | ??                                     | $0     | Borrow      | TMAC                                             |
| Screens                | x3 70" screens                         | $X     | Rent        | TMAC                                             |
| Tripod                 | Steel camera tripod or video tripid    | $0     | Own or Rent | @Shrinks99 (backup: @dcwalk) or Sheridan College |
| Camera                 | Canon XA10 Camcorder                   | $140   | Rent        | Vistek (Sheridan / York $0 rentals TBD)          |
| Capture Cards          | Elgato HD60 / Avermedia GL310 LGP Lite | $0     | Own         | @ASoTNetworks & @dcwalk                          |
| HDMI Splitter          | Keliiyo 1x4 active HDMI splitter       | $30    | Buy         | Amazon                                           |
| HDMI Monitor           | x2 Blackmagic Video Assist Monitor     | $210   | Rent        | Vistek                                           |
| SD Cards               | x4 SanDisk Ultra 128GB microSDXC UHS-I | $108   | Buy         | Amazon                                           |
| Streaming PC           | ??                                     | $0     | Own         | @ASoTNetworks's laptop                           |
| HDMI cables            | x2 25ft HDMI cables                    | $37.06 | Buy         | Amazon                                           |
| Offline Audio Recorder | Zoom H4N                               | $0     | Rent        | Sheridan College / York University               |

**Comment by benhylau** [comment]
patcon says: https://www.pubnub.com/developers/twitch-tv-obs-subtitles/

**Comment by benhylau** [comment]
Hank has thoughts on roles, will publish docs. Most important docs are roles: incl. team monitoring stream

**Comment by Shrinks99** [comment]
Scheduling for the actual day:  I am not free on Thursday until 4:30 and I'll be in Oakville at that time.  I'll likely be able to be in Toronto for 6:00 which unfortunately is the exact time the venue closes.  For the venue walkthrough I can only be available on Tuesday.  I would like to attend this if possible as I had to miss the preliminary one last year and that was somewhat unlucky.

I would like to get familiar with the camera before Thursday in order to be able to properly operate it.  Vistek offers various late night rentals and things so hopefully we can work something out there in order to not drive up cost.  I will be testing the cameras that York and Sheridan offer as backups. 
 I'd also like to run a test stream the day before so we know what to sort out.

According to @dcwalk the first stream is starting at 2:00 PM Friday (barring schedule changes and whatnot), of course ideally everything runs without issue by that time.

**Comment by benhylau** [comment]
Confirming my purchase of these from Canada Computers:

- 1 x [iCAN HDMI v2.0 1X4 Splitter](https://www.canadacomputers.com/product_info.php?cPath=5_1336_1338&item_id=107106) CAD 40 _(not the Keliiyo one, that one only supports HDMI v1.4)_
- 2 x [iCAN HDMI 28AWG Version 2.0 25 ft.](https://www.canadacomputers.com/product_info.php?cPath=5_1340_102&item_id=108094) CAD 50 _(@DarkDrgn2k do we have ones from last year, are they 25 ft.?)_
- 4 x [SanDisk Ultra 128GB microSDXC UHS-I](https://www.canadacomputers.com/product_info.php?cPath=11_175_177&item_id=121295) CAD 112

## Approvals

- [ ] @Shrinks99 
- [ ] @dcwalk

**Comment by dcwalk** [comment]
Confirming that purchase, but yes it would be nice to reuse the Cables from last year @darkdrgn2k

**Comment by darkdrgn2k** [comment]
* [ ]  IPFS livestream code changes
       
       * [ ]  Committed: Pure nginx ([PR and tasklist](tomeshnet/ipfs-live-streaming#76))
       * [ ]  Committed: Improve player stream selection UX (e.g. default to one stream if other is not available) --
       * [ ]  ~~Stretch: Chunk playlist archives at each hour~~ - Dont need to anymore
       * [ ]  Stretch: Scrollback from live player tomeshnet/ipfs-live-streaming#87
       * [ ]  Stretch: Tagging chunks live and tweet (stretch goal) -- tweet button on player
         ournetworks.ca only change tomeshnet/ipfs-live-streaming#88

**Comment by darkdrgn2k** [comment]
>     * 2 x [iCAN HDMI 28AWG Version 2.0 25 ft.](https://www.canadacomputers.com/product_info.php?cPath=5_1340_102&item_id=108094) CAD 50 _(@darkdrgn2k do we have ones from last year, are they 25 ft.?)_

I think they where 25ft but i'll confirm tonigh

I have both the cables, but i think one of them may not be functioning correctly.  I had issues using it so i would count on only 1.

**Comment by dcwalk** [comment]
> > ```
> > * 2 x [iCAN HDMI 28AWG Version 2.0 25 ft.](https://www.canadacomputers.com/product_info.php?cPath=5_1340_102&item_id=108094) CAD 50 _(@darkdrgn2k do we have ones from last year, are they 25 ft.?)_
> > ```
> 
> I think they where 25ft but i'll confirm tonigh
> 
> I have both the cables, but i think one of them may not be functioning correctly. I had issues using it so i would count on only 1.

Would still be good to have both at the event!

**Comment by darkdrgn2k** [comment]
> 
> 
> patcon says: https://www.pubnub.com/developers/twitch-tv-obs-subtitles/

I ran some of last years talks through this. 

If the person is talking CLEARLY and SLOWLY it works well

But it did not work very well with the recordings.

**Comment by dcwalk** [comment]
Thanks, replies inline: 

> Scheduling for the actual day: I am not free on Thursday until 4:30 and I'll be in Oakville at that time. I'll likely be able to be in Toronto for 6:00 which unfortunately is the exact time the venue closes. For the venue walkthrough I can only be available on Tuesday. I would like to attend this if possible as I had to miss the preliminary one last year and that was somewhat unlucky.
> 

What time on Tuesday? 
The walkthrough is for everything, not just A/V, so I can't confirm dates with all the moving piecies. I think we can commit to at least one person from the A/V team being able to attend.

> I would like to get familiar with the camera before Thursday in order to be able to properly operate it. Vistek offers various late night rentals and things so hopefully we can work something out there in order to not drive up cost. I will be testing the cameras that York and Sheridan offer as backups.

I'm not used to two bookings. If we rent gear Thursday night we have ~6 hours with it before we start. Would the full morning Friday not be enough to familiarize with the camera before we go live? 

> I'd also like to run a test stream the day before so we know what to sort out.

This is not going to be possible from the site if A/V folks are not able to be there for load-in. The testing that was done last year could be done again?

> According to @dcwalk the first stream is starting at 2:00 PM Friday (barring schedule changes and whatnot), of course ideally everything runs without issue by that time.

**Comment by benhylau** [comment]
Purchased all the equipment (with slight changes) and will pick up on Thurs and bring to tomesh meet for testing.

**Comment by benhylau** [comment]
We are expecting three sessions that will be either:

- Pre-recorded video presentation + Live Q&A
- Live video presentation + Live Q&A

In addition, we have presenters who indicated they want to be on the livestream, but not have the recording of their talk released afterwards (and vice versa). @darkdrgn2k our final long-running "live archive" m3u8 that we publish needs to have segments of chunks redacted. I think we also need to support the "stream-time-independent chunk-hash-based bookmarking" that we discussed.

**Comment by benhylau** [comment]
I copied the above equipment table and removed all the stuff we already own:

| *Equipment*            | *Description*                          | *Cost* | *Access*    | *Provider*                                       |
|------------------------|----------------------------------------|--------|-------------|--------------------------------------------------|
| Mixer                  | ??                                     | $0     | Borrow      | TMAC                                             |
| Microphones            | x4 ??                                  | $0     | Borrow      | TMAC                                             |
| Lectern                | ??                                     | $0     | Borrow      | TMAC                                             |
| Speakers               | ??                                     | $0     | Borrow      | TMAC                                             |
| Screens                | x3 70" screens                         | $X     | Rent        | TMAC                                             |
| Camera                 | Canon XA10 Camcorder                   | $140   | Rent        | Vistek (Sheridan / York $0 rentals TBD)          |
| HDMI Monitor           | x2 Blackmagic Video Assist Monitor     | $210   | Rent        | Vistek                                           |
| Offline Audio Recorder | Zoom H4N                               | $0     | Rent        | Sheridan College / York University               |

It seems the only thing we need to rent from Vistek are:
- Canon XA10 Camcorder $140 (not needed if we get one from Sheridan)
- x2 Blackmagic Video Assist Monitor $210

@Shrinks99 is this correct? Do we still need this if we rent the camera from Sheridan?

The TMAC and Sheridan rentals are okay?
- [ ] @dcwalk
- [ ] @Shrinks99

**Comment by Shrinks99** [comment]
@benhylau We will still need the Blackmagic monitors regardless, I will be testing the camera hopefully this week to see if we can rent it from Sheridan or York.

**Comment by benhylau** [comment]
Our rental for two [Blackmagic Design Video Assist 5" 1080p Monitor is confirmed at Vistek](
https://www.vistek.ca/rentals/details/R305499/blackmagic-design-video-assist-5-1080p-monitor)

We can pick them up on Thursday between 5:30 to 6:00 pm. If somehow that fails then 10:00 am on Friday.
We are to return them at 10:00 am on Monday.

The cost is CAD 240 + tax with a CAD 500 hold on my credit card.

**Comment by dcwalk** [comment]
These things happened! I'm closing as there aren't lingering TODOs. A new issue should be started for video post production
